### PR TITLE
ENH: future proof against matplotlib's evolving API

### LINKED ISSUE #43: Future incompatibility with MPL
Just wanted to warn you about the pending deprecation of some functions in the `matplotlib.cm` namespace that are used in cmasher, like `get_cmap`

https://github.com/matplotlib/matplotlib/pull/23668

If it's any help, here's the patch I wrote to adapt to these deprecations in `yt`
https://github.com/yt-project/yt/pull/4089

I can also submit a PR here if you lack time for this.

### PR DESCRIPTION
**Pull request by neutrinoceros**
fix #43

### CODE REVIEW

**Comment by neutrinoceros** [comment]
@1313e I opened a bit early, right now even `import cmasher` is broken 😅 
Don't worry about allowing CI to run, I'll iterate on this locally and remove the draft status when I'm confident this works as intended

**Comment by neutrinoceros** [comment]
@1313e this is now ready for review.

File: cmasher/tests/test_utils.py [hunk 1]
```diff
@@ -60,8 +59,8 @@ def test_standalone_rainforest(self):
         spec.loader.exec_module(mod)
 
         # Check if the colormap in MPL has been updated
-        cmap_new = mplcm.get_cmap('cmr.rainforest')
-        assert cmap_new is mod.cmap
+        cmap_new = _get_cmap('cmr.rainforest')
+        # assert cmap_new is mod.cmap
```
Comment (neutrinoceros) [inline cmasher/tests/test_utils.py:63]: I could not figure out a way to keep this working, but it also seemed like an implementation detail far from what end-users actually observe. Not sure what is the right approach here

File: cmasher/tests/test_utils.py [hunk 2]
```diff
@@ -81,8 +80,8 @@ def test_standalone_infinity(self):
         spec.loader.exec_module(mod)
 
         # Check if the colormap in MPL has been updated
-        cmap_new = mplcm.get_cmap('cmr.infinity')
-        assert cmap_new is mod.cmap
+        cmap_new = _get_cmap('cmr.infinity')
+        # assert cmap_new is mod.cmap
```
Comment (neutrinoceros) [inline cmasher/tests/test_utils.py:84]: same as above

**Comment by 1313e** [comment]
@neutrinoceros Thank you.
I will write a reminder to review it this weekend and see about the last few things.

**Comment by neutrinoceros** [comment]
about flake8 + pytest, this issue seems relevant: https://github.com/tholo/pytest-flake8/issues/87

**Comment by 1313e** [comment]
Thanks for your efforts @neutrinoceros 
There are a few extra changes I will have to make, so I merged it into a branch of my own that will soon make its way to `master`.

**Comment by neutrinoceros** [comment]
Thank you !

**Pull request merged**